tests/test_layout_updater.py

```diff
@@ -0,0 +1,37 @@
+from pathlib import Path
+
+import mock
+import dash
+
+from webviz_config.common_cache import CACHE
+from webviz_config.themes import default_theme
+from webviz_config.plugins import _syntax_highlighter
+
+
+def test_layout_updater(dash_duo):
+
+    app = dash.Dash(__name__)
+    app.config.suppress_callback_exceptions = True
+    CACHE.init_app(app.server)
+    code_file = Path("./tests/data/basic_example.yaml")
+    with mock.patch("webviz_config.plugins._syntax_highlighter.get_path") as mock_path:
+        mock_path.return_value = code_file
+        page = _syntax_highlighter.SyntaxHighlighter(app, code_file)
+        app.layout = page.layout
+        dash_duo.start_server(app)
+        # empty input dict should return the theme layout
+        assert default_theme.plotly_theme[
+            "layout"
+        ] == default_theme.update_plotly_layout({})
+        # empty input dict should return the theme layout
+        specified_layout = {
+            "colorway": ["#FFFFFF", "#0000FF"],
+            "xaxis": {"title": {"text": "Title"}, "type": "log", "constrain": "domain"},
+            "legend": {"font": {"family": "Arial"}},
+        }
+        # make a deep updated layout from specified and theme layouts
+        new_layout = default_theme.update_plotly_layout(specified_layout)
+        # test values
+        assert new_layout["colorway"] == specified_layout["colorway"]
+        assert new_layout["xaxis"]["title"]["text"] == "Title"
+        assert dash_duo.get_logs() == [], "browser console should contain no error"
```

Comment:
Looks good. Would also be nice with a very simple test (can be in the same file as this), where:

1. Create a new theme, where a simple `plotly` layout dictionary is attached.
1. Make a simple dictionary variable as an example of a "plugins `plotly` layout modificitaion".
1. Call `update_plotly_layout` with this variable. The output is checked/verified to be the expected merged dictionary.
1. The `plotly` theme inside the Webviz theme should be checked that is unmodified after the call.
1. The variable from step two should be checked that is unmodified by the `update_plotly_layout` call.